Reorganise mpl_toolkits documentation

jenshnielsen opened the pull request:
The mpl_toolkits axes_grid and axisartist documentation is very focused on the axes_grid namespace dispite the fact that this is not recommenced for use. The code has been split into axes_grid1 and axisartist. This PR aims to do the same for the documentation and make the documentation more visible and consistent.  

* Move related packages that we don't ship to external resources
* Reorganise axes_grid documentation to split in axes_grid1 and axisartist
* Split examples and figures into 2 folders.
* Move the documentation up one level in the toc to make it more obvious where the toolkit documentation is.

This is WIP because the text still needs some editing for consistency. A tiny bit of functionality is still hidden in axes_grid.parasiteaxis which I plan to more to axesartist.

I am also tempted to move deprecate axes_grid and interested in input to this.

The docs as build by this branch is up at http://jenshnielsen.github.io/matplotlib

tacaswell commented:
I am not a fan of putting the external toolkits under the same top-level heading.  Maybe add a top-level "Third-party toolkits" section?

jenshnielsen commented:
I not quite sure what you mean. Are you suggesting to move the page from `External Resources` to a separate top level heading?  That makes sense to me.

tacaswell commented:
Yes, something like:

```
 - external resources
 - MPL API
 - Toolkits
    - axes_gird1
    - mplot3d
    - 
 - Third-party extensions packages
    - trendvis
    - seaborn
    - ..
```

jenshnielsen commented:
:+1:

jenshnielsen commented:
I think this should be ready for review. The branch at http://jenshnielsen.github.io/matplotlib is updated with the result of building 156cb7f

WeatherGod commented:
Uhm, pretty sure this isn't the result of your changes, but the logo/banner
at the top of the homepage is completely messed up for me. The text of
"matplotlib" is too large causing the "m" and the "b" to be cut off on
either end the radar chart logo to be right on top of it.

Using Firefox 43 on Ubuntu.

On Tue, Dec 29, 2015 at 9:12 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> I think this should be ready for review. The branch at
> http://jenshnielsen.github.io/matplotlib is updated with the result of
> building 156cb7f
> <https://github.com/matplotlib/matplotlib/commit/156cb7fd11c43bff8d7d715ec88cc6d54dbfc2b4>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-167797993>
> .
>

jenshnielsen commented:
@WeatherGod Is that any different from http://matplotlib.org/devdocs/?

WeatherGod commented:
No, it is the same on both pages.

On Tue, Dec 29, 2015 at 9:41 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> @WeatherGod <https://github.com/WeatherGod> Is that any different from
> http://matplotlib.org/devdocs/?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-167802088>
> .
>

jenshnielsen commented:
It's probably due to https://github.com/matplotlib/matplotlib/pull/5653 then?

WeatherGod commented:
as for the mpl_toolkits page itself:
http://jenshnielsen.github.io/matplotlib/mpl_toolkits/index.html, it feels
like there is too much redundant stuff. For example, there is the "mplot3d"
section header, then there is the "mplot3d" bullet point, and then the
"Matplotlib mplot3d toolkit" sub-bullet point. The axes_grid1 and
axisartist stuff isn't too bad, but still feels a little redundant.

On Tue, Dec 29, 2015 at 9:12 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> I think this should be ready for review. The branch at
> http://jenshnielsen.github.io/matplotlib is updated with the result of
> building 156cb7f
> <https://github.com/matplotlib/matplotlib/commit/156cb7fd11c43bff8d7d715ec88cc6d54dbfc2b4>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-167797993>
> .
>

WeatherGod commented:
Most likely, I have posted a comment there.

On Tue, Dec 29, 2015 at 9:44 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> It's probably due to #5653
> <https://github.com/matplotlib/matplotlib/pull/5653> then?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-167802441>
> .
>

jenshnielsen commented:
@WeatherGod I agree that it's a bit redundant but I have not actually changed that around. That is as it has always been.

jenshnielsen commented:
But perhaps we should just remove all the TOCs from that page?

mdboom commented on doc/_templates/index.html:
```diff
@@ -156,13 +156,21 @@
+    extend and build on Matplotlib functionality,
+    including a choice of two projection and mapping toolkits
+    <a href="http://matplotlib.org/basemap">basemap</a> and
+    <a href="http://scitools.org.uk/cartopy/docs/latest">cartopy</a>
```
Needs `.`

mdboom replied:
Would it make sense to link to https://github.com/matplotlib here, since we are encouraging most third-party packages to come under our gh organization?

mdboom commented on doc/mpl_toolkits/axisartist/overview.rst:
```diff
@@ -0,0 +1,117 @@
+
+.. plot:: mpl_toolkits/axisartist/examples/demo_floating_axis.py
+
+Since it uses a special artists, some mpl commands that work on
```
`uses a special` -> `uses special`

mdboom commented on doc/mpl_toolkits/axisartist/overview.rst:
```diff
@@ -0,0 +1,117 @@
+
+*axisartist* module provides a custom (and very experimental) Axes
+class, where each axis (left, right, top and bottom) have a separate
+artist associated which is responsible to draw axis-line, ticks,
```
`artist associated` -> `associated artist`

mdboom commented on doc/mpl_toolkits/axisartist/overview.rst:
```diff
@@ -0,0 +1,117 @@
+in the data coordinate (i.e., the axis floats around when viewlimit
+changes).
+
+The axes class, by default, have its xaxis and yaxis invisible, and
```
`have` -> `has`

mdboom commented on doc/mpl_toolkits/axisartist/overview.rst:
```diff
@@ -0,0 +1,117 @@
+changes).
+
+The axes class, by default, have its xaxis and yaxis invisible, and
+has 4 additional artists which are responsible to draw axis in
```
`to draw axis` -> `to draw the 4 axis sides`

mdboom commented:
Oops -- sorry, I missed that this is still a WIP and the text hasn't been edited.  I'll hold off on my grammatical/typographical nitpicks until that's done.

jenshnielsen commented:
Is not WIP any more but by the looks of it most of the typos are in the original text which I have just moved.

mdboom commented:
I see.  Should we try to fix those now anyway?  I'm happy to just do them as a PR against this one if that's more efficient...

jenshnielsen commented:
We might as well fix obvious typos now. I have tried not to do to many heavy edits but most of the examples could definitely need an edit but the question is how much effort we want to put into this part of the code?

mdboom commented:
I'll spend a few minutes (but no more) fixing obvious typos and submit a PR against this one...

mdboom commented:
Actually, now that I'm understanding the size of this better, I think it's probably best to leave the text as-is for this PR and try to clean it up later.  (We really need to devote some real effort to editing the docs for grammar/consistency in a lot of places...)

jenshnielsen commented:
@WeatherGod I have removed the toc trees

@mdboom I corrected the typos that you already noted and added a note above the matplotlib organization to thirdpartypackages/index

jenshnielsen commented:
I have added an api changes noted that `axes_grid` is deprecated.

The few files which were different in `axes_grid` have been copied to `axisartist` 
They are mainly Axes from `axes_grid` but using `axisartist` axis. This should really be refactored away to make the Axes take an axis type on construction.

WeatherGod commented:
Has the github.io been updated for the latest changes?

On Wed, Dec 30, 2015 at 7:46 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> I have added an api changes noted that axes_grid is deprecated.
>
> The few files which were different in axes_grid have been copied to
> axisartist
> They are mainly Axes from axes_grid but using axisartist axis. This
> should really be refactored away to make the Axes take an axis type on
> construction.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-167991239>
> .
>

jenshnielsen commented:
Sorry I was on a flaky internet connection on the train should be there now

WeatherGod commented:
looking much better now.

On Wed, Dec 30, 2015 at 11:36 AM, Jens Hedegaard Nielsen <
notifications@github.com> wrote:

> Sorry I was on a flaky internet connection on the train should be there now
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5752#issuecomment-168029740>
> .
>

jenshnielsen commented:
Rebased onto current master

jenshnielsen commented:
I have uploaded a new preview version to http://jenshnielsen.github.io/matplotlibdocs rather than http://jenshnielsen.github.io/matplotlib

jenshnielsen commented:
I think this should be ready to go. The documentation is by no means perfect but I think this is a significant structural improvement.

WeatherGod commented on doc/api/api_changes/2015-12-30-JHN.rst:
```diff
@@ -0,0 +1,7 @@
+All functionallity from `mpl_toolkits.axes_grid` can be found in either 
+`mpl_toolkits.axes_grid1` or `mpl_toolkits.axisartist`. Axes classes from 
+`mpl_toolkits.axes_grid` based on `Axis` from `mpl_toolkits.axisartist` can be
+fund in `mpl_toolkits.axisartist`
```
"fund" --> "found"

jenshnielsen commented:
By the way. I'm happy to revet the deprecation of `axes_grid` and just change the docs in this PR

WeatherGod commented on doc/mpl_toolkits/index.rst:
```diff
@@ -77,34 +31,28 @@ solution for some use cases.
-The matplotlib :ref:`AxesGrid <toolkit_axesgrid-index>` toolkit is a collection of helper classes to
-ease displaying multiple images in matplotlib. The AxesGrid toolkit is
-distributed with matplotlib source.
+The :ref:`mpl_toolkits.axisgrid1 <toolkit_axesgrid1-index>` toolkit is a
```
"axisgrid1"?

jenshnielsen replied:
Good catch that is embarrassing

WeatherGod commented on doc/thirdpartypackages/index.rst:
```diff
@@ -0,0 +1,130 @@
+=======================
+
+`matplotlib_iterm2 <https://github.com/oselivanov/matplotlib_iterm2>`_ is an
+external matplotlib backend uses iTerm2 nightly build inline image display
```
uses --> using

WeatherGod commented:
I think that's all the typos I can find. I vote for deprecation of axes_grid. It has been a long time coming, and splitting axisartist out in the documentation makes things a lot more understandable.

WeatherGod commented:
This has a +1 from me. Should this be backported to v2.x or should it stay in master?

jenshnielsen commented:
I was not planning to backport it to 2.x but I can be convinced otherwise. Any objections to merging this. It moves around a large number of examples so I would like to merge it before it acquires to many conflicts.

WeatherGod commented:
ok, we will leave it in master for now
